scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -279,185 +240,226 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
-      // If a comma is followed by a new line & then a closing paren, bracket or brace
-      // then it is a trailing comma and is ignored.
       def isTrailingComma: Boolean =
         dialect.allowTrailingCommas &&
           curr.is[Comma] &&
           next.is[CloseDelim] &&
           next.pos.startLine > curr.pos.endLine
-      if (curr.isNot[Trivia] && !isTrailingComma) {
-
-        var sepRegions =
-          if (indentedRegion(sepRegionsParameter) && currentIndent >= 0) {
-            var sepRegionsProcess = sepRegionsParameter
-            while ((indentedRegion(sepRegionsProcess)
-                && sepRegionsProcess.head.indent > currentIndent && !isLeadingInfixOperator(
-                  curr
-                ) && !prev.is[CanContinueOnNextLine]) ||
-              shouldCloseCaseOnNonCase(sepRegionsProcess)) {
-              insertOutdent()
-              // match can start an identation, block but if `match` follows it means it's chaining matches
-              if (curr.isNot[KwMatch]) {
-                parserTokens += new LF(curr.input, curr.dialect, curr.start)
-                parserTokenPositions += currPos
-              }
-              sepRegionsProcess = sepRegionsProcess.tail
-            }
-            sepRegionsProcess
-          } else sepRegionsParameter
-
-        sepRegions = if (expectedIndent > 0 && dialect.allowSignificantIndentation) {
-          if (!sepRegions.isEmpty && sepRegions.head == RegionEnumArtificialMark)
-            RegionIndentEnum(expectedIndent) :: sepRegions.tail
-          else RegionIndent(expectedIndent, prev.is[KwMatch]) :: sepRegions
-        } else if (shouldCloseIndent &&
-          sepRegions.headOption.exists(_.isInstanceOf[RegionIndent])) {
-          sepRegions.tail
-        } else {
-          sepRegions
-        }
 
-        var sepRegionsNew = {
-          if (curr.is[LeftParen]) RegionParen :: sepRegions
-          else if (curr.is[LeftBracket]) RegionBracket :: sepRegions
-          else if (curr.is[Comma] &&
-            sepRegions.headOption.exists(_.isInstanceOf[RegionIndent]) &&
-            sepRegions.tail.headOption.contains(RegionParen)) {
-            insertOutdent()
-            sepRegions.tail
-          } else if (curr.is[LeftBrace]) {
-            val indentInBrace = if (isAheadNewLine(currPos)) countIndent(nextPos) else -1
-            // After encountering keyword Enum we add artificial '{' on top of stack.
-            // Then always after Enum next token is '{'. On token '{' we check if top of stack is '{'
-            // (which in case of enum is always true) and replace it with '$'.
-            // Now if we have token 'case' and top of stack is '$' we know it is Enum-case.
-            // In any other case it is 'match-case' or 'try-case'
+      def mkIndent: TokenRef =
+        TokenRef(
+          new Indentation.Indent(curr.input, curr.dialect, curr.start, curr.end),
+          prevPos,
+          currPos
+        )
+
+      def mkOutdent: TokenRef =
```

Comment:
```suggestion
      def prependOutdent: TokenRef =
```
?

Reply:
In my sense `prepend`  doesn't fit there. The new implementation can't prepend, it only returns the next token based on the current state.

If you are unsure about shortened `make` what if it will be `makeOutdent`/`createOutdent` or maybe just `outdent`?

Reply:
Just a suggestion, you can ignore me really here.